src/weights.jl

```diff
@@ -1,9 +1,9 @@
 
 ###### Weight vector #####
 
-immutable WeightVec{W,Vec<:RealVector}
-    values::Vec
-    sum::W
+immutable WeightVec{T, V<:RealVector, S} <: RealVector{T}
```

Comment:
My view was that all of these could be useful in the cases like `WeightVec{Bool, NullableArray{Bool}, Int}`, but I figured you'd be more likely to want to dispatch on `V` than `S`.